partial/libuptiny/crypto_api.h

```diff
@@ -32,23 +32,25 @@ typedef struct {
   uint8_t sig[CRYPTO_MAX_SIGNATURE_LEN];
 } crypto_key_and_signature_t;
 
-typedef struct {
-  size_t bytes_fed;
-  uint8_t block[SHA512_BLOCK_SIZE];
-  struct sha512_state sha_state;
-  const uint8_t* signature;
-  const uint8_t* pub;
-} crypto_verify_ctx_t;
+typedef struct crypto_verify_ctx crypto_verify_ctx_t;
+typedef struct crypto_hash_ctx crypto_hash_ctx_t;
 
 void crypto_verify_init(crypto_verify_ctx_t* ctx, crypto_key_and_signature_t* sig);
 void crypto_verify_feed(crypto_verify_ctx_t* ctx, const uint8_t* data, size_t len);
 bool crypto_verify_result(crypto_verify_ctx_t* ctx);
 
+void crypto_hash_init(crypto_hash_ctx_t* ctx);
+void crypto_hash_feed(crypto_hash_ctx_t* ctx, const uint8_t* data, size_t len);
+void crypto_hash_result(crypto_hash_ctx_t* ctx, crypto_hash_t* hash);
+
 crypto_algorithm_t crypto_str_to_keytype(const char* keytype, size_t len);
 int crypto_get_keylen(crypto_algorithm_t alg);
```

Comment:
Should return a `size_t`? Also, change the fields in `alg_match_t` and `hash_match_t` to the same?